synoptic/src/synoptic/model/export/JsonExporter.java

```diff
@@ -96,151 +110,326 @@ public EventInstance(int traceID, int eventIndexWithinTrace) {
         }
     }
 
+    // Map of nodes and their respective global ID's
+    static Map<EvBasedNode, Integer> nodesIDMap = new LinkedHashMap<EvBasedNode, Integer>();
+
     /**
-     * Creates the 'log' of the JSON object: a list of traces within the log of
-     * this partition graph
+     * Creates the 'nodes' of the JSON object: a list of nodes within this
+     * EvBasedGraph
      * 
-     * @param pGraph
-     *            The partition graph whose log we're outputting
+     * @param evGraph
+     *            The EvBasedGraph whose nodes we're outputting
      */
-    private static List<Map<String, Object>> makeLogJSON(
-            PartitionGraph pGraph) {
-        // The log (list of traces) to go into the JSON object
-        List<Map<String, Object>> logListOfTraces = new LinkedList<Map<String, Object>>();
-
-        // Get all partitions in the partition graph
-        Set<Partition> allPartitions = pGraph.getNodes();
-
-        // Get the INITIAL partition, which will be used to retrieve all traces
-        // and their events
-        Partition initialPart = null;
-        for (Partition part : allPartitions) {
-            if (part.isInitial()) {
-                initialPart = part;
-                break;
+    private static List<Map<String, Integer>> makeNodesJSON(
+            EvBasedGraph evGraph) {
+
+        // List of Nodes to go into the JSON object
+        List<Map<String, Integer>> nodeList = new LinkedList<Map<String, Integer>>();
+
+        // add the initial node first
+        Map<String, Integer> initialNodeMap = new LinkedHashMap<String, Integer>();
+        initialNodeMap.put("id", globalID);
+        nodesIDMap.put(evGraph.getInitialNode(), globalID);
+        initialNodeMap.put("displayableIDs: ",
+                displayablesIDMap.get("INITIAL"));
+        nodeList.add(initialNodeMap);
+        globalID++;
```

Comment:
You perform the same four operations for the initial node, then all other nodes, then the terminal node: (1) make single node map and set "id", (2) store in nodesIDMap, (3) add displayable if initial or terminal only, (4) increment globalID. Extract this into a method, something that you would call with something like:
```
Map<String, Integer> singleNodeMap = makeNode(evGraph.getInitialNode(), initialStr);
nodeList.add(singleNodeMap);
```
And then do these same calls for the nodes in the loop and for the terminal node. Make sure your method can handle *not* adding a displayable for normal nodes.